taglocker_test.go

```diff
@@ -0,0 +1,162 @@
+package plc
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+)
+
+const alphaCharset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+const numCharset = "0123456789"
+const alphanumCharset = alphaCharset + "_" + numCharset
+
+// NoOpReadWriter is a dummy ReadWriter interface that unconditionally succeeds.
+type NoOpReadWriter struct {
+	state uint32
```

Comment:
If you don't actually use state, may as well leave it out: `type NoOpReadWriter struct {}`.
You could also use the `DeviceFake ` in `device_test.go`, but it does more than you need, so your NoOp is better. (In an in-review branch I've renamed `DeviceFake` to `FakeReadWriter`.)

Reply:
State gets used in the ReadTag and WriteTag methods just so the race detector has something to potentially catch.  Happy to still take it out.

Reply:
It's accessed/written in those functions, but it looks like it's little more than a NOP, so I figured it was leftover code.
Maybe you could comment it in the struct? e.g. `// state is present to detect race conditions with the -race flag` Then some ignorant developer like me won't come along and delete it. 😀